pynxtools_xps/vms/vamas.py

```diff
@@ -490,157 +448,102 @@ def _read_lines(self, file: Union[str, Path]):
-        block.month = int(self.data.pop(0).strip())
-        block.day = int(self.data.pop(0).strip())
-        block.hour = int(self.data.pop(0).strip())
-        block.minute = int(self.data.pop(0).strip())
-        block.second = int(self.data.pop(0).strip().split(".")[0])
-        block.no_hrs_in_advance_of_gmt = int(self.data.pop(0).strip())
-        block.no_comment_lines = int(self.data.pop(0).strip())
-        for _ in range(block.no_comment_lines):
-            block.comment_lines += [self.data.pop(0)]
-        block.technique = self.data.pop(0).strip()
-        for _ in range(int(self.header.nr_exp_var)):
-            block.exp_var_value = self.data.pop(0).strip()
-        block.source_label = self.data.pop(0).strip()
-        block.source_energy = float(self.data.pop(0).strip())
-        block.unknown_1 = self.data.pop(0).strip()
-        block.unknown_2 = self.data.pop(0).strip()
-        block.unknown_3 = self.data.pop(0).strip()
-        block.source_analyzer_angle = float(self.data.pop(0).strip())
-        block.unknown_4 = self.data.pop(0).strip()
-        block.analyzer_mode = self.data.pop(0).strip()
-        block.resolution = float(self.data.pop(0).strip())
-        block.magnification = self.data.pop(0).strip()
-        block.work_function = float(self.data.pop(0).strip())
-        block.target_bias = float(self.data.pop(0).strip())
-        block.analyzer_width_x = float(self.data.pop(0).strip())
-        block.analyzer_width_y = float(self.data.pop(0).strip())
-        block.analyzer_take_off_polar_angle = float(self.data.pop(0).strip())
-        block.analyzer_azimuth = float(self.data.pop(0).strip())
-        block.species_label = self.data.pop(0).strip()
-        block.transition_label = self.data.pop(0).strip()
-        block.particle_charge = int(self.data.pop(0).strip())
-
-        if self.header.scan_mode == "REGULAR":
-            block.abscissa_label = self.data.pop(0).strip()
-            block.abscissa_units = self.data.pop(0).strip()
-            block.abscissa_start = float(self.data.pop(0).strip())
-            block.abscissa_step = float(self.data.pop(0).strip())
+        for exp_var_no in range(int(self.header.num_exp_var)):
+            exp_var = ExpVariable()
+            for attr in ["label", "unit"]:
+                self._setattr_with_datacls_type(exp_var, attr, self.data.pop(0).strip())
+                setattr(
+                    self.header, f"exp_var_{exp_var_no}_{attr}", getattr(exp_var, attr)
+                )
 
-            block.no_variables = int(self.data.pop(0).strip())
-            for var in range(block.no_variables):
-                name = "variable_label_" + str(var + 1)
-                setattr(block, name, self.data.pop(0).strip())
-                name = "variable_units_" + str(var + 1)
-                setattr(block, name, self.data.pop(0).strip())
+        self.header.num_entries_in_inclusion_list = int(self.data.pop(0).strip())
```

Comment:
name indicates that it is a list but getting int. Meaning full name is helpful for others.

Reply:
This is the number of entries in the inclusion list. I kept it like this because this is how the standard defines it.